wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -1108,10 +1110,39 @@ private void reportJoystickUnpluggedWarning(String message) {
     double currentTime = Timer.getFPGATimestamp();
     if (currentTime > m_nextMessageTime) {
       reportWarning(message, false);
+      if (m_silenceJoystickWarnings) {
+        m_nextMessageTime = Double.POSITIVE_INFINITY;
+        return;
+      }
       m_nextMessageTime = currentTime + JOYSTICK_UNPLUGGED_MESSAGE_INTERVAL;
     }
   }
 
+  /**
+   * Sets whether to silence the DriverStation warning prints when a Joystick is unplugged.
+   *
+   * @param silenced true to enable, false to disable silencing.
+   */
+  public void silenceJoystickUnplugged(boolean silenced) {
```

Comment:
For symmetry with the getter function, this should probably use "suppress" instead of "silence". Same with the member variable.